src/pages/salesProducts/SalesProducts.jsx

```diff
@@ -0,0 +1,133 @@
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import useResize from "../../lib/hooks/useResize";
+
+import SalesProductCardList from "../../components/salesProducts/SalesProductCardList";
+import Title from "../../core/titles/Title";
+import fetchProduct from "../../lib/api/product";
+
+import "./sales-products.css";
+import SearchInput from "../../core/search/SearchInput";
+import Dropdown from "../../core/dropdown/Dropdown";
+import BtnSmall from "../../core/buttons/BtnSmall";
+import PagiNation from "../../components/salesProducts/PagiNation";
+
+const SalesProducts = () => {
+  const [salesProducts, setSalesProducts] = useState([]);
+  const [order, setOrder] = useState("recent");
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalPage, setTotalPage] = useState(0);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  const getSalesPageSize = (width) => {
+    if (1200 <= width) {
+      return 10;
+    }
+    if (768 <= width) {
+      return 6;
+    }
+    return 4;
+  };
+  const size = useResize({ getPageSize: getSalesPageSize });
+
+  const handleSalesProducts = async (options) => {
+    let result;
+    try {
+      setIsLoading(true);
+      setErrorMessage(null);
+      result = await fetchProduct(options);
+    } catch (error) {
+      setErrorMessage(error);
+      return;
+    } finally {
+      setIsLoading(false);
+    }
+    const { list, totalCount } = result;
+    setSalesProducts(list);
+    const newTotalPage = Math.ceil(totalCount / size);
+    setTotalPage(newTotalPage);
```

Comment:
에러일때 강제로 리턴해서 아랫줄이 실행되는걸 막기보다는 try 문안에 둠으로써 자연스러운 흐름을 가져 갈 수 있을 것 같아요.

```suggestion
    try {
      setIsLoading(true);
      setErrorMessage(null);
      result = await fetchProduct(options);
      const { list, totalCount } = result;
      setSalesProducts(list);

    } catch (error) {
      setErrorMessage(error);
    } finally {
      setIsLoading(false);
    }
    const newTotalPage = Math.ceil(totalCount / size);
    setTotalPage(newTotalPage);
```